highlighters/main/main-highlighter.zsh

```diff
@@ -432,6 +468,65 @@ _zsh_highlight_highlighter_main_paint()
       fi
     fi
 
+    # Expand aliases.
+    # TODO: this should be done iteratively, e.g., 'alias x=y y=z z=w\n x' 
+    #       And then the entire 'alias' branch of the 'case' statement should
+    #       be done here.
+    # TODO: path expansion should happen _after_ alias expansion
+    _zsh_highlight_main_highlighter_expand_path $arg
+    _zsh_highlight_main__type "$REPLY"
+    local res="$REPLY"
+    if [[ $res == "alias" ]]; then
+      _zsh_highlight_main__resolve_alias $arg
+      () {
+        # Use a temporary array to ensure the subscript is interpreted as
+        # an array subscript, not as a scalar subscript
+        local -a reply
+        # TODO: the ${interactive_comments+set} path needs to skip comments; see test-data/alias-comment1.zsh
+        reply=( ${interactive_comments-${(z)REPLY}}
+                ${interactive_comments+${(zZ+c+)REPLY}} )
+        arg=$reply[1]
+      }
+    fi
+
+    # Expand parameters.
+    #
+    # ### For now, expand just '$foo' or '${foo}', possibly with braces, but with
+    # ### no other features of the parameter expansion syntax.  (No ${(x)foo},
+    # ### no ${foo[x]}, no ${foo:-x}.)
+    () {
+      # That's not entirely correct --- if the parameter's value happens to be a reserved
+      # word, the parameter expansion will be highlighted as a reserved word --- but that
+      # incorrectness is outweighed by the usability improvement of permitting the use of
+      # parameters that refer to commands, functions, and builtins.
+      local -a match mbegin mend
+      local MATCH; integer MBEGIN MEND
+      local parameter_name
+      if [[ $arg[1] == '$' ]] && [[ ${arg[2]} == '{' ]] && [[ ${arg[-1]} == '}' ]]; then
+        parameter_name=${${arg:2}%?}
+      elif [[ $arg[1] == '$' ]]; then
+        parameter_name=${arg:1}
+      fi
+      if [[ $res == none ]] && zmodload -e zsh/parameter &&
+         [[ ${parameter_name} =~ ^([A-Za-z_][A-Za-z0-9_]*|[0-9]+)$ ]] &&
+         (( ${+parameters[${MATCH}]} ))
+         then
+        # Set $arg.
+        case ${(tP)MATCH} in
+          (*array*|*assoc*)
+            local -a words=( ${(P)MATCH} )
```

Comment:
This will error on pre-5.1.

Reply:
Fixed, thanks.

(I spotted this late yesterday but ran into PEBKAGH trying to record it here)